core/src/main/java/com/linecorp/armeria/server/logging/AccessLogComponent.java

```diff
@@ -138,47 +147,43 @@ static DateTimeFormatter findFormatter(@Nullable String variable) {
             if (variable == null) {
                 return defaultDateTimeFormatter;
             }
-            // Find pre-defined formatter first.
-            switch (variable) {
-                case "BASIC_ISO_DATE":
-                    return DateTimeFormatter.BASIC_ISO_DATE;
-                case "ISO_LOCAL_DATE":
-                    return DateTimeFormatter.ISO_LOCAL_DATE;
-                case "ISO_OFFSET_DATE":
-                    return DateTimeFormatter.ISO_OFFSET_DATE;
-                case "ISO_DATE":
-                    return DateTimeFormatter.ISO_DATE;
-                case "ISO_LOCAL_TIME":
-                    return DateTimeFormatter.ISO_LOCAL_TIME;
-                case "ISO_OFFSET_TIME":
-                    return DateTimeFormatter.ISO_OFFSET_TIME;
-                case "ISO_TIME":
-                    return DateTimeFormatter.ISO_TIME;
-                case "ISO_LOCAL_DATE_TIME":
-                    return DateTimeFormatter.ISO_LOCAL_DATE_TIME;
-                case "ISO_OFFSET_DATE_TIME":
-                    return DateTimeFormatter.ISO_OFFSET_DATE_TIME;
-                case "ISO_ZONED_DATE_TIME":
-                    return DateTimeFormatter.ISO_ZONED_DATE_TIME;
-                case "ISO_DATE_TIME":
-                    return DateTimeFormatter.ISO_DATE_TIME;
-                case "ISO_ORDINAL_DATE":
-                    return DateTimeFormatter.ISO_ORDINAL_DATE;
-                case "ISO_WEEK_DATE":
-                    return DateTimeFormatter.ISO_WEEK_DATE;
-                case "ISO_INSTANT":
-                    return DateTimeFormatter.ISO_INSTANT;
-                case "RFC_1123_DATE_TIME":
-                    return DateTimeFormatter.RFC_1123_DATE_TIME;
-                default:
-                    try {
-                        return DateTimeFormatter.ofPattern(variable, Locale.ENGLISH);
-                    } catch (Throwable cause) {
-                        throw new IllegalArgumentException("unexpected date/time format variable: " +
-                                                           variable, cause);
-                    }
+
+            if (builtinFormatters == null) {
+                initializePredefinedFormatters();
```

Comment:
Perhaps we don't really need to initialize this lazily?

Reply:
Ok, got it!